chrono-tz/src/timezone_impl.rs

```diff
@@ -403,3 +404,69 @@ impl TimeZone for Tz {
         TzOffset::new(*self, timespans.get(index))
     }
 }
+
+/// Represents the information of a gap.
+pub struct GapInfo {
+    /// When available it contains information about the beginning of the gap.
+    ///
+    /// The time represents the first instant in which the gap starts.
+    /// This means that it is the first instant that when used with [`TimeZone::from_local_datetime`]
+    /// it will return [`LocalResult::None`].
+    ///
+    /// The offset represents the offset of the first instant before the gap.
+    pub begin: Option<(NaiveDateTime, TzOffset)>,
+    /// When available it contains the first instant after the gap.
+    pub end: Option<DateTime<Tz>>,
+}
+
+impl Tz {
+    /// Returns information about a gap.
+    ///
+    /// It returns `None` if `local` is not in a gap for the current timezone.
+    ///
+    /// If `local` is at the limits of the known timestamps the fields `begin` or `end` in
+    /// [`GapInfo`] will be `None`.
+    pub fn gap_info_from_local_datetime(&self, local: &NaiveDateTime) -> Option<GapInfo> {
```

Comment:
Don't like this name very much. Maybe this should just be a constructor for `GapInfo`, like `GapInfo::new(dt: NaiveDateTime, tz: &Tz)`?

(I think we should take `NaiveDateTime` by value.)

Reply:
I change it to `GapInfo::new`. I was happy with the name either but I wasn't able to come up with something better.
Regarding how to take `NaiveDateTime` I think it should be by reference because I expect this to be used when `TimeZone::from_local_datetime` returns `LocalResult::None` and `TimeZone::from_local_datetime` takes it as reference. No strong opinion so happy to change if you disagree.

Reply:
That's fair, I'm okay with leaving it as a reference.